compiler/src/codegen.rs

```diff
@@ -329,7 +319,7 @@ impl<'a> MethodGen<'a> {
         match self.method_type().0 {
             // Unary
             MethodType::Unary => {
-                pub_async_fn(w, &self.unary_async(&method_name), |w| {
+                pub_async_fn(w, &self.unary(&method_name), |w| {
```

Comment:
Not compatible with old ttrpc versions, we need a major version release.

Reply:
This change is API compatible, since a `&mut T` could be safely coerced to a `&T`, and will not generate compile errors on existing code AFAIK. (However, a warning may be generated, since most existing code uses `let mut` to define a `Client`, which is now redundant.)

But I agree that this change could be postponed to the next major version, anyway.

Reply:
![image](https://user-images.githubusercontent.com/1013713/147047538-78e218f5-6c4c-454f-bc23-c11bf74bc200.png)

It's not compatible, this error would occur.